application-forum-ui/src/main/resources/ForumCode/ForumSheet.xml

```diff
@@ -55,14 +55,15 @@
   #elseif($xcontext.action == 'view')
     #initForum($spaceName)
     ## Depending on XE version, we consider parent-child relationship for Pre NS or space-subspace for NS.
-    #set($hierarchyCondition = "doc.parent=:forumDoc or doc.space=:forumSpace")
-    #set($topics = $services.query.xwql("from doc.object(ForumCode.TopicClass) as topic where $hierarchyCondition").bindValue('forumDoc', $doc.fullName).bindValue('forumSpace', $doc.space).execute())
+    #set($hierarchyCondition = 'doc.parent=:parent or doc.space=:space')
```

Comment:
`doc.space=:space` means documents in the forum space. In NS, a topic is a subspace, not a subdocument in the forum space. So `doc.space` in this case is actually the topic space. You need the parent of the topic, so you could use something like `doc.space LIKE :spacePrefix` (and make sure to bind spacePrefix to `"${doc.space}.%"`).